### PR TITLE
User Can See Drink details

### PR DESCRIPTION
**Pull request by kaylawoodbury**
# [PT Story](https://www.pivotaltracker.com/story/show/171702599)
## Changes proposed in this pull request:
* Create test for displaying drink details
* Create fixture file for margarita drink details
* Add an button for drink details
* Add functionality to App. jsx to display drink details when the detail button is clicked
## What I have learned working on this feature:
* How to map within a map
* How to pull an id from a previous array response, and send it back to the back end as a GET request

## Screenshots
Screenshot 2020-03-14 at 21 00 33
Screenshot 2020-03-14 at 21 00 50

### CODE REVIEW

File: cypress/integration/userCanSeeDrinkDetails.feature.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      url: "http://localhost:3000/api/v1/cocktails/**",
+      response: "fixture:margarita_details.json"
+    });
+    cy.get("#details-button").click();
+    // cy.get("#details")
+    //   .invoke("attr", "src")
+    //   .should(
+    //     "include",
+    //     "https://www.thecocktaildb.com/images/media/drink/wpxpvu1439905379.jpg"
+    //   );
```
Comment (tochman) [inline cypress/integration/userCanSeeDrinkDetails.feature.js]: Please remove comments

File: src/App.jsx [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
 import axios from "axios";
-
 class App extends Component {
   state = {
     query: "",
     results: [],
-    message: ""
+    message: "",
+    details: [],
+    details_error_message: "",
```
Comment (tochman) [inline src/App.jsx]: 1. Do we need a specific state for this message 2. Please use camelCase in JavaScript

File: src/App.jsx [hunk 3]
```diff
@@ -28,16 +27,37 @@ class App extends Component {
...
+                <br />
+                <div id="drink-name" key={item.id} data-id={item.id}>{item.name}
+                  <button
+                    id="details-button"
+                    onClick={this.seeDetails.bind(this)}
+                    key={item.id}
+                  >
+                    Details
+                </button></div>
+                {item.category} {item.IBA} <br />
```
Comment (tochman) [inline src/App.jsx]: I would extract this to a separate component but it can be done as a refactoring ticket

File: src/App.jsx [hunk 4]
```diff
@@ -46,7 +66,30 @@ class App extends Component {
...
+                    <div key={item.id}>
+                      {content.name} {content.measure}
+                    </div>
+                  )
+                })}
+                Instruction: {item.instructions}
+              </div>
+            );
+          })}
+        </div>
```
Comment (tochman) [inline src/App.jsx]: Same here, we can refactor this to a separate component

File: src/App.jsx (hunk 3 above)
Comment (kaylawoodbury) [reply src/App.jsx]: Yes, we are planning to refactor everything in a separate PR to clean it up after we get the main functionality built out.

File: cypress/integration/userCanSeeDrinkDetails.feature.js (hunk 1 above)
Comment (kaylawoodbury) [reply cypress/integration/userCanSeeDrinkDetails.feature.js]: removed

File: src/App.jsx (hunk 2 above)
Comment (kaylawoodbury) [reply src/App.jsx]: Removed this as well since we weren't actually using it

**Pull request merged**